src/charm.py

```diff
@@ -121,14 +146,17 @@ def _on_install(self, _):
         # Patch/create Kubernetes resources
         try:
             self.unit.status = MaintenanceStatus("Creating auth resources")
-            self._create_resource(resource_type="auth", context=self._context)
+            self._create_auth()
             self.unit.status = MaintenanceStatus("Creating CRDs")
             self._create_crds()
         except ApiError as e:
             logging.error(traceback.format_exc())
             self.unit.status = BlockedStatus(
                 f"Creating/patching resources failed with code {str(e.status.code)}."
             )
+            if e.status.code == 403:
+                event.defer()
```

Comment:
I don't have experience with `.defer()` so I don't know when is a good/bad time to use it.  The awkward part is that it just reruns this event again immediately before whatever the next event is, when that event happens.  

I think that doing a `juju trust` probably triggers at least a config-changed hook, so you might not need this at all.

Reply:
> The awkward part is that it just reruns this event again immediately before whatever the next event is

In this case this behaviour is what we want because no other event handler creates the resources, waiting for config-changed will only cause more issues, like `404s` because it will try to patch something that doesn't exist. I decided to use `defer()` to avoid adding more logic to the `config_changed` event handler like: patch if exists, if not, create.